visualization/src/resource-visualization.cc

```diff
@@ -256,31 +306,55 @@ void visualizeReprojectedDepthResource(
           return;
         }
 
+        if (FLAGS_vis_pointcloud_filter_dense_map_before_publishing) {
+          pcl::PointCloud<pcl::PointXYZI>::Ptr pcl_cloud(
+              new pcl::PointCloud<pcl::PointXYZI>);
+          CHECK_NOTNULL(pcl_cloud);
+          backend::convertPointCloudType(points_S, &(*pcl_cloud));
+          /*
+          voxelGridPointCloud(
+              FLAGS_vis_pointcloud_filter_leaf_size_m, pcl_cloud);
+          backend::convertPointCloudType((*pcl_cloud), &points_S);
+          */
+        }
+
         // Either publish in local frame with a tf or in global frame.
         if (FLAGS_vis_pointcloud_publish_in_sensor_frame_with_tf) {
+          sensor_msgs::PointCloud2 ros_point_cloud_S;
           if (FLAGS_vis_pointcloud_color_random) {
             resources::PointCloud points_C_colorized = points_S;
             accumulated_point_cloud_G.colorizePointCloud(r, g, b);
             sensor_msgs::PointCloud2 ros_point_cloud_S;
             backend::convertPointCloudType(
                 points_C_colorized, &ros_point_cloud_S);
-            publishPointCloudInLocalFrameWithTf(T_G_S, &ros_point_cloud_S);
           } else {
-            sensor_msgs::PointCloud2 ros_point_cloud_S;
             backend::convertPointCloudType(points_S, &ros_point_cloud_S);
-            publishPointCloudInLocalFrameWithTf(T_G_S, &ros_point_cloud_S);
           }
+          if (FLAGS_vis_pointcloud_filter_dense_map_before_publishing) {
+            applyVoxelGridFilter(ros_point_cloud_S, &ros_point_cloud_S);
+          } else if (
+              FLAGS_vis_pointcloud_filter_beautify_dense_map_before_publishing) {  // NOLINT
+            applyBeautification(points_S, &ros_point_cloud_S);
```

Comment:
Additionally the flow of the entire `applyBeautification` function is unclear to me, it seems to drop the content of `ros_point_cloud_S` totally and just replaces it based on `points_S`. Then all the previous work is useless always